plugins/pulp_rpm/plugins/distributors/export_distributor/generate_iso.py

```diff
@@ -87,9 +88,37 @@ def _make_iso(file_list, target_dir, output_dir, filename):
     if status != 0:
         log.error("Error creating iso %s; status code: %d; output: %s" % (file_path, status, out))
     else:
+        _add_digest_file(file_path)
         log.info('Successfully created iso %s' % file_path)
 
 
+def _add_digest_file(file_path):
+    """
+    Parses each bit of the iso to generate a *.iso.DIGESTS file.
+
+    :param file_path: The file path to the ISO image file
+    :param file_path: str
+    """
+    exportedISO = open(file_path, "rb")
+    digestFile = open(file_path + ".DIGESTS", "w+")
+    hashers = {}
+    for hasher in hashlib.algorithms_guaranteed:
+        hashers[hasher] = getattr(hashlib, hasher)()
```

Comment:
This will fail on a FIPS environment. I think this should work:

```python
from pulp.server.util import md5 as pulp_md5

# ...

if hasher == 'md5':
    hashers[hasher] = pulp_md5()
else:
    hashers[hasher] = getattr(hashlib, hasher)()

Reply:
@daviddavis thanks :+1: now the pr is FIPS compliant too :)